lib/upload_creator.rb

```diff
@@ -44,14 +45,22 @@ def create_for(user_id)
     is_image ||= @image_info && FileHelper.is_supported_image?("test.#{@image_info.type}")
     is_image = false if @opts[:for_theme]
 
+    # if this is present then it means we are creating an upload record from
+    # an external_upload_stub and the file is > ExternalUploadManager::DOWNLOAD_LIMIT,
+    # so we have not downloaded it to a tempfile. no modifications can be made to the
+    # file in this case because it does not exist; we simply move it to its new location
+    # in S3
+    external_upload_too_big = @opts[:external_upload_too_big]
+    sha1_before_changes = Upload.generate_digest(@file) if @file
+
     DistributedMutex.synchronize("upload_#{user_id}_#{@filename}") do
       # We need to convert HEIFs early because FastImage does not consider them as images
-      if convert_heif_to_jpeg?
+      if convert_heif_to_jpeg? && !external_upload_too_big
```

Comment:
I'm not sure I like all these checks for `!external_upload_too_big`, may refactor this in future. There is just so little we can do for these huge external uploads that we haven't downloaded

Reply:
Yea I have the same feeling here. I wonder if we should just split it out into two different methods or just two different code paths.